### DISCUSSION TITLE
can i use /public folder as image storage?

### DISCUSSION THREAD

**Discussion by ruru-m07** [vercel/next.js#64541] [Help]
### Summary

i try to store user profile pics in `/public` folder 

## use case
- let's say i have user called `ruru-m07` and he upload profile image and i store that image in `/public/ruru-m07.png` that help me to fetch fetch image with just a user name

## ref example
- i have a github acout and my username is `ruru-m07` that visible in [https://github.com/ruru-m07](https://github.com/ruru-m07) but if you add `.png` in end you see my profile picture like this [https://github.com/ruru-m07.png](https://github.com/ruru-m07.png)

### Example

https://github.com/ruru-m07.png

**Comment by ruru-m07** [comment]
i want to done this thing with app router and Latest `v14.2.1`

**Comment by Icegreeen** [comment]
> ### Summary
> i try to store user profile pics in `/public` folder
> 
> ## use case
> * let's say i have user called `ruru-m07` and he upload profile image and i store that image in `/public/ruru-m07.png` that help me to fetch fetch image with just a user name
> 
> ## ref example
> * i have a github acout and my username is `ruru-m07` that visible in https://github.com/ruru-m07 but if you add `.png` in end you see my profile picture like this https://github.com/ruru-m07.png
> 
> ### Additional information
> _No response_
> 
> ### Example
> https://github.com/ruru-m07.png

The `/public` folder in Next.js is a special folder that is served statically. This means its contents are publicly accessible and can be referenced directly in web pages. By placing files, such as images, in the `/public` folder, Next.js makes them accessible at a URL relative to the root of your site.

 It's important to note that while using the `/public` folder is a convenient way to serve static files in a Next.js application, it may not be the best option for storing and serving user images at large scale. Depending on the needs of your application and the expected volume of user images.

**Reply by ruru-m07** [reply]
is any way to get images like `/ruru-m07.png` to image and `/ruru-m07` to user page 

see this 3rd line comment [gist](https://gist.github.com/joeschoe/7939469465ce44a3f533b1690ed12dd2#file-next-firebase-image-upoad-ts-L3)

**Reply by Icegreeen** [reply]
Okay, I read it. But what is your real objective, I need to understand the entire scope to see if the suggestion makes sense.

As I said, the /public folder is used to store public images that can be accessed through the project URL, landing page images for example.

Tell us what your project is about and what your ultimate goal is

**Reply by GabenGar** [reply]
He wants to use that folder to add images at runtime, which is not possible since it's only consumed at build-time.

**Comment by icyJoseph** [comment]
It is better to use a proper image storage service:

- https://github.com/vercel/next.js/discussions/15089#discussioncomment-36357
- https://github.com/orgs/vercel/discussions/241#discussioncomment-363002
- https://vercel.com/guides/how-can-i-use-files-in-serverless-functions

**Reply by ruru-m07** [reply]
### @icyJoseph really really useful info `respect`. 

can you suggest a example to write file in `public `folder? with nextjs app router.

# more detail 
- in `/` route we provide a input file that accept images and
- a button called `upload` 
- on click that button write image in `/public` folder.

#### @Icegreeen you really have `Galaxy Brain`
